### PR TITLE
zh, en: add `is-sharding` config item

### PR DESCRIPTION
**Pull request by GMHDBJD**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-dm) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)
add missing config item

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by lance6716** [comment]
`is-sharding` has lower priority than `shard-mode`, maybe we could only address `shard-mode` has three values and default is `""`

**Comment by Liuxiaozhen12** [comment]
`is-sharding` seems like a prerequisite for `shard-mode`. I don't understand why `is-sharding` has lower priority than `shard-mode`. So three values seem a little bit confusing to me. @lance6716

**Comment by lance6716** [comment]
> `is-sharding` seems like a prerequisite for `shard-mode`. I don't understand why `is-sharding` has lower priority than `shard-mode`. So three values seem a little bit confusing to me. @lance6716

currently `is-sharding` is an alias for `shard-mode = "pessimistic"` if not set `shard-mode`, and `shard-mode` is the actually setting that takes effect. `is-sharding` is a historical field which in my mind it should be deprecated. 🤔

**Comment by Liuxiaozhen12** [comment]
> > `is-sharding` seems like a prerequisite for `shard-mode`. I don't understand why `is-sharding` has lower priority than `shard-mode`. So three values seem a little bit confusing to me. @lance6716
> 
> currently `is-sharding` is an alias for `shard-mode = "pessimistic"` if not set `shard-mode`, and `shard-mode` is the actually setting that takes effect. `is-sharding` is a historical field which in my mind it should be deprecated. 🤔

PTAL~ @GMHDBJD

**Review by lance6716 — APPROVED** [review]

File: en/task-configuration-file-full.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ The following is the task configuration file template which allows you to perfor
 ## ********* Basic configuration *********
 name: test                      # The name of the task. Should be globally unique.
 task-mode: all                  # The task mode. Can be set to `full`(only migrates full data)/`incremental`(replicates binlog synchronously)/`all` (replicates both full and incremental binlogs).
-shard-mode: "pessimistic"       # This needs to be configured if it is a shard merge task. The "pessimistic" mode is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set to the "optimistic" mode.
+shard-mode: "pessimistic"       # The shard merge mode. Available modes are ""/"pessimistic"/"optimistic". The "" mode is used by default. If it's a shard merge task, please set to the "pessimistic" mode. After understanding the principles and restrictions of the "optimistic" mode, you can set to the "optimistic" mode.
```
Comment (Liuxiaozhen12) [inline en/task-configuration-file-full.md:31]:
```suggestion
shard-mode: "pessimistic"       # The shard merge mode. Optional modes are ""/"pessimistic"/"optimistic". The "" mode is used by default. If the task is a shard merge task, set it to the "pessimistic" mode. After understanding the principles and restrictions of the "optimistic" mode, you can set it to the "optimistic" mode.
```

**Review by Liuxiaozhen12 — APPROVED** [review]
rest LGTM

**Pull request merged**